tests/spec/Backend/TaggedCacheSpec.php

```diff
@@ -0,0 +1,172 @@
+<?php
+namespace spec\Cmp\Cache\Backend;
+
+use Cmp\Cache\Backend\TaggedCache;
+use Cmp\Cache\Cache;
+use Cmp\Cache\Exceptions\NotFoundException;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+/**
+ * Class TaggedCacheSpec
+ *
+ * @package spec\Cmp\Cache\Backend
+ * @mixin TaggedCache
+ */
+class TaggedCacheSpec extends ObjectBehavior
+{
+    private $uid = 'dummy_uid';
+
+    private $tagKey = 'tag:dummy';
+    
+    function let(Cache $store)
+    {
+        $this->beConstructedWith($store, 'dummy');
+        $store->get($this->tagKey)->willReturn($this->uid);
+        $store->set($this->tagKey, $this->uid)->willReturn(true);
+    }
+
+    function it_generates_namespace_key_correctly(Cache $store)
+    {
+        $store
+            ->get($this->tagKey)
+            ->willReturn()
+            ->shouldBeCalled();
+
+        $store
+            ->set($this->tagKey, Argument::type('string'))
+            ->willReturn(true)
+            ->shouldBeCalled();
+
+        $store
+            ->get(Argument::containingString('sux'))
+            ->willReturn(false)
+            ->shouldBeCalled();
```

Comment:
I think `shouldBeCalled` is not necessary, because `willReturn` already sets the expectation, but it doesn't hurt either and states clearer the intention